include/uprotocol-cpp-ulink-zenoh/message/messageParser.h

```diff
@@ -49,16 +49,18 @@ struct TLV {
 class MessageParser
 {
     public:
-        
+        static MessageParser& instance() noexcept;
```

Comment:
why do you need an singleton ?

Reply:
Using a singleton on MessageParser just makes the parsing rules consistent. I'll remove it for now if you think it is not needed.